### ISSUE TITLE
node-gyp is updated to v.9.3.0 according to the instruction but npm still uses node-gyp@3.8.0

### ISSUE THREAD

**Issue by astFuser** [nodejs/node-gyp#2761]
Hi NodeJS Team!

Again and again I have `gyp ERR! node-gyp -v v3.8.0` in the cli output for `npm install`.

**1** - I've updated `node-gyp` version according to the instruction here => https://github.com/nodejs/node-gyp/tree/main/docs#readme as follows:
```
$ npm explore npm/node_modules/@npmcli/run-script -g -- npm_config_global=false npm install node-gyp@latest
```
Unfortunately this did not help.

**2** - Then I updated `node-sass` globaly `npm install --global node-sass@latest` despite the fact that I have `sass` in my project - no luck.
**3** - I've modified `npm` config and set up `node_gyp` there too:
```
node_gyp = "/home/user/.nvm/versions/node/v18.12.1/lib/node_modules/node-gyp/bin/node-gyp.js"
```
**4** - I tried to clean npm cache
**5** - tried to rebuild `node-gyp`

All the above did not help me to update my `node-gyp` for my  `npm install`.

Could you please help or point me in the right direction here. Any help is really appreciated.

* **Node Version**: `v18.12.1`
* **Npm Version**: `8.9.12`
* **node-gyp Version**: `v9.3.0`
* **Python Version**: `Python 3.10.6`
* **Platform**: `Linux user-System-Product-Name 5.15.0-52-generic #58-Ubuntu SMP Thu Oct 13 08:03:55 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* **Compiler**:`Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/11/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.3.0-1ubuntu1~22.04' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-11-xKiWfi/gcc-11-11.3.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-11-xKiWfi/gcc-11-11.3.0/debian/tmp-gcn/usr --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.3.0 (Ubuntu 11.3.0-1ubuntu1~22.04)`
* **Module**: `npm install` locally

#### npm install log:
```
$ npm install                                                                                
npm WARN config init.author.email Use `--init-author-email` instead.
npm WARN config init.author.name Use `--init-author-name` instead.
npm WARN config init.license Use `--init-license` instead.
npm WARN deprecated ini@1.3.5: Please update to ini >=1.3.6 to avoid a prototype pollution issue
npm WARN deprecated set-value@0.2.0: Critical bug fixed in v3.0.1, please upgrade to the latest version.
npm WARN deprecated set-value@0.2.0: Critical bug fixed in v3.0.1, please upgrade to the latest version.
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm WARN deprecated @stylelint/postcss-markdown@0.36.1: Use the original unforked package instead: postcss-markdown
npm WARN deprecated har-validator@5.1.3: this library is no longer supported
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated request-promise@4.2.6: request-promise has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated debug@4.2.0: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@1.7.0: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated chokidar@1.7.0: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated source-map-url@0.4.0: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated xmldom@0.1.31: Deprecated due to CVE-2021-21366 resolved in 0.5.0
npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated axios@0.19.0: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
npm WARN deprecated istanbul@0.4.5: This module is no longer maintained, try this instead:
npm WARN deprecated   npm i nyc
npm WARN deprecated Visit https://istanbul.js.org/integrations for other alternatives.
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated webdriverio@4.14.4: outdated version, please use @next
npm WARN deprecated flag-icon-css@2.9.0: The project has been renamed to flag-icons
npm WARN deprecated core-js@2.6.11: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@3.18.0: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path /home/user/fd-ecomm/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c -- node scripts/build.js
npm ERR! Building: /home/user/.nvm/versions/node/v18.12.1/bin/node /home/user/fd-ecomm/node_modules/node-sass/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   '/home/user/.nvm/versions/node/v18.12.1/bin/node',
npm ERR! gyp verb cli   '/home/user/fd-ecomm/node_modules/node-sass/node_modules/node-gyp/bin/node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@18.12.1 | linux | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "python3" in the PATH
npm ERR! gyp verb `which` succeeded python3 /usr/bin/python3
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: Command failed: /usr/bin/python3 -c import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack   File "<string>", line 1
npm ERR! gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
npm ERR! gyp ERR! stack SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)?
npm ERR! gyp ERR! stack 
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:412:12)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! gyp ERR! System Linux 5.15.0-52-generic
npm ERR! gyp ERR! command "/home/user/.nvm/versions/node/v18.12.1/bin/node" "/home/user/fd-ecomm/node_modules/node-sass/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /home/user/fd-ecomm/node_modules/node-sass
npm ERR! gyp ERR! node -v v18.12.1
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok 
npm ERR! Build failed with error code: 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-11-09T14_58_45_978Z-debug-0.log
```
#### npm config list:
```
$ npm config list
npm WARN config init.author.email Use `--init-author-email` instead.
npm WARN config init.author.name Use `--init-author-name` instead.
npm WARN config init.license Use `--init-license` instead.
; "user" config from /home/user/.npmrc

[--global] = "devdir" 
@fortawesome:registry = "https://npm.fontawesome.com/" 
//npm.fontawesome.com/:_authToken = (protected) 
//registry.npmjs.org/:_authToken = (protected) 
/tmp/.gyp = "" 
init.author.email = "user@gmail.com" 
init.author.name = "user" 
init.license = "MIT" 
msvs_version = "2017" 
node_gyp = "/home/user/.nvm/versions/node/v18.12.1/lib/node_modules/node-gyp/bin/node-gyp.js" 
python = "python3" 

; "env" config from environment

devdir = "/tmp/.gyp" 

; node bin location = /home/user/.nvm/versions/node/v18.12.1/bin/node
; node version = v18.12.1
; npm local prefix = /home/user/fd-ecomm
; npm version = 8.19.2
; cwd = /home/user/fd-ecomm
; HOME = /home/user
; Run `npm config ls -l` to show all defaults.
```
#### node-gyp rebuild
```
$ node-gyp rebuild
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.0
gyp info using node@18.12.1 | linux | x64
gyp info find Python using Python version 3.10.6 found at "/usr/bin/python3"
gyp http GET https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v18.12.1/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v18.12.1/SHASUMS256.txt
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/home/fuser/.nvm/versions/node/v18.12.1/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/fuser/ASTOUND/FLOORANDDECOR/fd-ecomm/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/fuser/.nvm/versions/node/v18.12.1/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/tmp/.gyp/18.12.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/tmp/.gyp/18.12.1',
gyp info spawn args   '-Dnode_gyp_dir=/home/fuser/.nvm/versions/node/v18.12.1/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/tmp/.gyp/18.12.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/fuser/ASTOUND/FLOORANDDECOR/fd-ecomm',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/fuser/ASTOUND/FLOORANDDECOR/fd-ecomm/build'
  SOLINK_MODULE(target) Release/obj.target/binding.node
  COPY Release/binding.node
make: Leaving directory '/home/fuser/ASTOUND/FLOORANDDECOR/fd-ecomm/build'
gyp info ok 
```

**Comment by cclauss** [comment]
Note that `node-sass` (deprecated) and `sass` are different and you probably want the latter.
% `npm search sass`
```
NAME                      | DESCRIPTION          | AUTHOR          | DATE       | VERSION  | KEYWORDS
sass                      | A pure JavaScript…   | =nex3 =xzyfer…  | 2022-11-09 | 1.56.1   | style scss sass preprocessor css
node-sass                 | Wrapper around…      | =saperski…      | 2022-11-09 | 8.0.0    | css libsass preprocessor sass scss style
[ ... ]
```

**Comment by astFuser** [comment]
I've tried with the version 1.56.1 for `sass` and had the same result for `npm install` as I posted above unfortunately.

**Comment by astFuser** [comment]
Here is my `package.json` file. Maybe it can be helpful here.

```
{
  "name": "user_project",
  "version": "1.0.0",
  "private": true,
  "engines": {
    "node": "18.12.1"
  },
  "dependencies": {
    "@fancyapps/ui": "^4.0.0-alpha.4",
    "@googlemaps/js-api-loader": "^1.12.5",
    "app-module-path": "^2.2.0",
    "bootstrap": "^4.5.2",
    "chai-subset": "^1.6.0",
    "cleave.js": "^1.6.0",
    "credit-card-type": "^9.1.0",
    "ebook-convert": "^2.0.1",
    "element-closest-polyfill": "^1.0.2",
    "eslint": "^6.8.0",
    "eslint-restricted-globals": "^0.2.0",
    "eventemitter3": "^3.1.0",
    "flag-icon-css": "^2.3.0",
    "font-awesome": "^4.6.3",
    "google-libphonenumber": "^3.2.13",
    "icon-font-generator": "^2.1.11",
    "intl-tel-input": "^14.0.6",
    "jquery": "^3.5.1",
    "js-datepicker": "^5.16.0",
    "nouislider": "^14.6.2",
    "photoswipe": "^4.1.3",
    "pinch-zoom-js": "^2.3.4",
    "plyr": "^3.6.8",
    "properties-parser": "^0.3.1",
    "request-promise": "^4.2.6",
    "sgmf-scripts": "^2.4.1",
    "signature_pad": "^3.0.0-beta.4",
    "smoothscroll-polyfill": "^0.4.4",
    "squirrelly": "^7.0.3",
    "tingle.js": "^0.15.3",
    "tiny-slider": "^2.9.3",
    "validator": "^12.1.0",
    "what-input": "^5.2.10",
    "youtube-iframe": "^1.0.3",
    "zingtouch": "^1.0.6"
  },
  "devDependencies": {
    "@babel/core": "^7.11.6",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-proposal-optional-chaining": "^7.16.7",
    "@babel/plugin-syntax-dynamic-import": "^7.2.0",
    "@babel/plugin-transform-runtime": "^7.11.0",
    "@babel/polyfill": "^7.11.5",
    "@babel/preset-env": "^7.11.5",
    "@babel/runtime": "^7.11.0",
    "ansi-colors": "^4.1.1",
    "atlas-guide": "^2.2.4",
    "autoprefixer": "^9.8.6",
    "babel-eslint": "^10.0.1",
    "babel-loader": "^8.0.6",
    "babel-minify": "^0.5.1",
    "browser-sync": "^2.26.7",
    "chai": "^4.2.0",
    "cheerio": "0.22.0",
    "core-js": "^2.6.3",
    "css-hot-loader": "^1.4.4",
    "css-loader": "^5.0.1",
    "diff-dom": "^4.2.0",
    "dwdav": "^3.4.1",
    "dwupload": "^3.8.2",
    "eslint": "^6.8.0",
    "eslint-config-airbnb-base": "^14.2.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-sitegenesis": "^1.0.0",
    "eslint-restricted-globals": "^0.2.0",
    "fancy-log": "^1.3.3",
    "globby": "^9.0.0",
    "gulp": "^4.0.0",
    "gulp-eslint": "^5.0.0",
    "gulp-header": "^2.0.9",
    "gulp-mustache": "^4.1.1",
    "gulp-postcss": "^8.0.0",
    "gulp-pug": "^4.0.1",
    "gulp-rename": "^1.4.0",
    "gulp-sass": "^4.0.2",
    "gulp-sourcemaps": "^2.6.4",
    "gulp-template": "^5.0.0",
    "gulp.spritesmith": "^6.10.1",
    "inputmask": "^5.0.5",
    "istanbul": "^0.4.4",
    "jira-client": "^8.0.0",
    "lnk": "^1.1.0",
    "merge-stream": "^2.0.0",
    "mini-css-extract-plugin": "^1.3.4",
    "mocha": "^8.1.3",
    "npm-run-all": "^4.1.5",
    "popper.js": "^1.16.1",
    "postcss-loader": "^3.0.0",
    "rimraf": "^2.6.3",
    "sanitize-filename": "^1.6.3",
    "sass": "^1.56.1",
    "sass-graph": "^3.0.5",
    "sass-loader": "^10.1.1",
    "scandit-sdk": "^5.7.1",
    "set-value": "^3.0.2",
    "sfcc-ci": "^2.9.1",
    "shelljs": "^0.8.4",
    "simple-parallax-js": "^5.6.2",
    "sinon": "^7.2.2",
    "slim-select": "^1.26.0",
    "stylelint": "^13.7.2",
    "stylelint-config-recommended": "^3.0.0",
    "stylelint-config-standard": "^20.0.0",
    "stylelint-order": "^4.1.0",
    "stylelint-scss": "^3.18.0",
    "terser-webpack-plugin": "^5.3.6",
    "tingle": "0.0.0",
    "tippy.js": "^6.3.7",
    "vanilla-lazyload": "^12.5.1",
    "vinyl-buffer": "^1.0.1",
    "webpack": "^5.74.0",
    "webpack-cli": "^4.10.0",
    "webpack-dev-middleware": "^5.3.3",
    "webpack-dev-server": "^4.11.1",
    "webpack-format-messages": "^2.0.6",
    "webpack-hot-middleware": "^2.25.0",
    "whatwg-fetch": "^3.4.1",
    "yargs": "^17.2.1"
  }
}

```

**Comment by BrunoFenzl** [comment]
I think the underlying issue is similar to mine (https://github.com/nodejs/node-gyp/issues/2760).

**Comment by cclauss** [comment]
Please add the line [`"nodes-gyp": "^9.3.0",`](https://github.com/nodejs/node-gyp/releases) to the `devDependencies` of your `package.json` file and try again.

**Comment by astFuser** [comment]
@cclauss  thanks for help!
done.
I added `"node-gyp": "^9.3.0"` to the `devDependencies` and tried `npm install`.
Unfortunately this did not help either ((( 
the issue is still the same - `npm ERR! gyp ERR! node-gyp -v v3.8.0`. 
The output in the cli with the error was the same too.

**Comment by astFuser** [comment]
Hi @cclauss !

done.

- folder `/home/user/.cache/node-gyp` was deleted,
- `node_modules` and `package-lock.json` were deleted too,

- `<%user%>/.electron-gyp` is not present on my linux system.
here is the output below:
```
$ ls -lah ~/.electron 
total 46M
drwxrwxr-x   2 fuser fuser 4.0K Jun 29  2019 .
drwxr-xr-x 155 fuser fuser  12K Nov 10 13:45 ..
-rw-rw-r--   1 fuser fuser  46M Jun 29  2019 electron-v1.8.8-linux-x64.zip
-rw-rw-r--   1 fuser fuser 5.7K Jun 29  2019 SHASUMS256.txt-1.8.8
```

The above did not help 😔

I tried a next comment too in the thread - https://github.com/nodejs/node-gyp/issues/2714#issuecomment-1309376926
```
$ npx node-gyp install
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.0
gyp info using node@18.12.1 | linux | x64
gyp http GET https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v18.12.1/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v18.12.1/SHASUMS256.txt
undefined
gyp info ok
```
The installation shows that the version is node-gyp@9.3.0.
BUT `npm install` still shows the initial error - `npm ERR! gyp ERR! node-gyp -v v3.8.0` 🤕

**Comment by cclauss** [comment]
Sorry...  Can you try #2272

**Comment by astFuser** [comment]
Hi @cclauss !

thanks a lot for all your help!

The reason was a dependency inside an old `gulp-sass` package.
Its version was `4.14.1`. I updated it to the version `5.1.0` and the issue was resolved.

here is an excerpt from the log:

find this one in `/home/user/.npm/_logs/2022-11-10T12_49_11_324Z-debug-0.log`

```
3131 timing idealTree:node_modules/gulp-rename Completed in 0ms
3132 silly placeDep ROOT node-sass@4.14.1 OK for: gulp-sass@4.1.1 want: ^4.8.3
3133 silly placeDep node_modules/gulp-sass strip-ansi@4.0.0 OK for: gulp-sass@4.1.1 want: ^4.0.0
```
as we can see it wanted `^4.8.3` but I added the latest one - `5.1.0` and it was ok too.
